src/libraries/System.Net.Sockets/src/System/Net/Sockets/NetworkStream.cs

```diff
@@ -264,7 +264,17 @@ public override int Read(Span<byte> buffer)
             ThrowIfDisposed();
             if (!CanRead) throw new InvalidOperationException(SR.net_writeonlystream);
 
-            int bytesRead = _streamSocket.Receive(buffer, SocketFlags.None, out SocketError errorCode);
+            int bytesRead;
+            SocketError errorCode;
+            try
+            {
+                bytesRead = _streamSocket.Receive(buffer, SocketFlags.None, out errorCode);
+            }
+            catch (Exception exception) when (!(exception is OutOfMemoryException))
+            {
+                throw GetCustomNetworkException(SR.Format(SR.net_io_writefailure, exception.Message), exception);
```

Comment:
Presumably this should be `SR.net_io_writefailure`, right? Similarly below.

Reply:
Fixed.